.github/workflows/build.yml

```diff
@@ -42,9 +42,18 @@ jobs:
     runs-on: ubuntu-latest
     needs: compile
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/download-artifact@v2
+      - uses: actions/checkout@v4
+      - uses: actions/download-artifact@v3
         with:
           name: ubuntu
       - run: ./run-tests.sh
         working-directory: tests
+
+  unittest:
```

Comment:
Run the unit tests on a PR push.
Rather than wait for the build and then pull the artifacts, since I set up the unittests as a 'game' directory (in hopes of being able to make a very simple setup), it was more sensible to just do this since it doesn't reuse the executables in any way.